src/pfs/pfs.proto

```diff
@@ -196,6 +196,9 @@ message ListRepoRequest {
 message DeleteRepoRequest {
   Repo repo = 1;
   bool force = 2;
+  // If project present, and if repo is absent, then all repos in
+  // project will be deleted.
+  Project project = 3;
```

Comment:
since Repo already contains project, shouldn't we rely on that? If we want to express deleting an entire project, I think that should be its own RPC

Reply:
> since Repo already contains project, shouldn't we rely on that?

My first attempt did that: a request to delete a repo with an empty repo name would delete all repos in that repo’s project. But on consideration that seemed like a footgun, so I figured it was better to be more explicit.

There’s already an RPC to delete a project, `DeleteProject`; `pachctl delete repo --all` is specifically intended to delete all repos within a single project. Are you advocating a separate RPC `DeleteRepos`, which takes a request containing a project?

Reply:
Can we do a one_of, specify a Repo or a Project?  If you give a Repo, we use that, if you specify a project we'll use that and find all the Repos?  There would be no ambiguous behavior, and force could be outside the one_of.

Reply:
> Can we do a one_of, specify a Repo or a Project?

Oh, that's interesting. The [protobuf docs do specify](https://developers.google.com/protocol-buffers/docs/proto3#oneof) ‘you can safely move a single field into a new oneof.’

Reply:
Ah, but the Go gRPC bindings will change incompatibly.

Maybe it’s cleaner just to add the separate RPC `DeleteRepos` instead and not try to cram multi-repo deletion into `DeleteRepo` after all?

Reply:
That makes sense to me.